django/cantusdb_project/cantusindex.py

```diff
@@ -0,0 +1,150 @@
+"""
+A collection of functions for fetching data from
+Cantus Index's (CI's) various APIs.
+"""
+
+import requests
+from typing import Optional, Union
+from main_app.models import Genre
+
+CANTUS_INDEX_DOMAIN: str = "https://cantusindex.uwaterloo.ca"
+DEFAULT_TIMEOUT: float = 2  # seconds
+NUMBER_OF_SUGGESTED_CHANTS: int = 3  # this number can't be too large,
+# since for each suggested chant, we make a request to Cantus Index.
+# We haven't yet parallelized this process, so setting this number
+# too high will cause the Chant Create page to take a very long time
+# to load. If/when we parallelize this process, we want to limit
+# the size of the burst of requests sent to CantusIndex.
+
+
+def get_suggested_chants(
+    cantus_id: str, number_of_suggestions: int = NUMBER_OF_SUGGESTED_CHANTS
+) -> Optional[list[dict]]:
+    endpoint_path: str = f"/json-nextchants/{cantus_id}"
+    all_suggestions: Optional[list] = get_json_from_ci_api(endpoint_path)
+
+    if not all_suggestions:
+        # get_json_from_ci_api timed out
+        # or CI returned a response with no suggestions.
+        return None
+
+    # when Cantus ID doesn't exist within CI, CI's api returns a 200 response with `['Cantus ID is not valid']`
+    first_suggestion = all_suggestions[0]
+    if not isinstance(first_suggestion, dict):
+        return None
+
+    sort_by_occurrences: function = lambda suggestion: int(suggestion["count"])
```

Comment:
`function` is not a valid type. 
```suggestion
    sort_by_occurrences: Callable[[dict], int] = lambda suggestion: int(suggestion["count"])
```